setup.py

```diff
@@ -21,6 +21,19 @@
 site.ENABLE_USER_SITE = "--user" in sys.argv[1:]
 
 
+def cpp_path():
+    return os.environ.get("CC", "cpp")
+
+
+def cpp_args(args=[]):
+    cc = os.environ.get("CC")
+
+    if cc is not None:
+        return ["-E"] + args
+    else:
+        return args
+
```

Comment:
Can't we just pass "-E" directly to cpp_args regardless of if $CC is set?

Reply:
clang's cpp implementation will complain:

cpp: warning: argument unused during compilation: '-E' [-Wunused-command-line-argument]

Reply:
I can live with the warning, but I'll leave it to you

Reply:
The less unnecessary warnings the better, but its not a big deal. If you dislike the optional parameter I can remove it - but then you you want to push the option directly into the preprocess_file?

Reply:
Lets go with your current version